include/wifi_hal_generic.h

```diff
@@ -602,336 +744,442 @@ typedef enum {
-     wifi_interface_name_idex_map_t interface_map[(MAX_NUM_RADIOS * MAX_NUM_VAP_PER_RADIO)];
-     radio_interface_mapping_t radio_interface_map[MAX_NUM_RADIOS];
-     BOOL radio_presence[MAX_NUM_RADIOS];         /**< Indicates if the interfaces is present (not in deep sleep)*/
-}__attribute__((packed)) wifi_platform_property_t;
+typedef enum
+{
+    WIFI_BAND_NONE = 0x1, /**< No band. */
+    WIFI_BAND_2_5 = 0x2, /**< 2.4GHz band. */
+    WIFI_BAND_2_6 = 0x4, /**< 2.4GHz and 6GHz bands. */
+    WIFI_BAND_5_6 = 0x8, /**< 5GHz and 6GHz bands. */
+    WIFI_BAND_2_5_6 = 0x10, /**< 2.4GHz, 5GHz, and 6GHz bands. */
+    WIFI_BAND_2_5L = 0x20, /**< 2.4GHz and 5GHz low bands. */
+    WIFI_BAND_2_5H = 0x40, /**< 2.4GHz and 5GHz high bands. */
+    WIFI_BAND_5L_5H = 0x80, /**< 5GHz low and high bands. */
+    WIFI_BAND_2_5L_5H = 0x100 /**< 2.4GHz, 5GHz low, and 5GHz high bands. */
+} wifi_multi_link_bands_t;
 
 /**
- * @brief Wifi HAL Capabilities
+ * @brief Wi-Fi 7 supported modes.
  */
-typedef struct {
-    wifi_hal_version_t  version;            /**< The HAL version. */
-    wifi_platform_property_t wifi_prop;     /**< The platform Property that includes the number of radios and supported frequency bands. */
-    BOOL BandSteeringSupported;             /**< If BandSteeringSupported is TRUE, band-steering is support by the HAL */
-}__attribute__((packed)) wifi_hal_capability_t;
+typedef enum
+{
+    STR = 0x1, /**< Single-user Transmit and Receive (STR). */
+    NSTR = 0x2, /**< Non-STR. */
+    eMLSR = 0x4, /**< Enhanced Multi-Link Single-user Resource (eMLSR). */
+    eMLMR = 0x8 /**< Enhanced Multi-Link Multi-user Resource (eMLMR). */
+} wifi_multi_link_modes_t;
 
 /**
- * @brief Wifi TWT agreement type
+ * @brief Wi-Fi Multi-Link information.
  */
-typedef enum {
-    wifi_twt_agreement_type_individual,      /**< Set an individual TWT session. */
-    wifi_twt_agreement_type_broadcast,       /**< Set a Broadcast TWT session. */
+typedef struct _wifi_multi_link_info_t
+{
+    wifi_multi_link_bands_t mu_bands; /**< Supported bands for Multi-Link operation. */
+    wifi_multi_link_modes_t mu_modes; /**< Supported modes for Multi-Link operation. */
+} wifi_multi_link_info_t;
+
+/**
+ * @brief Wi-Fi platform properties.
+ */
+typedef struct
+{
+    UINT numRadios; /**< Number of radios. */
+    wifi_radio_capabilities_t radiocap[MAX_NUM_RADIOS]; /**< Radio capabilities. */
+    wifi_interface_name_idex_map_t interface_map[(MAX_NUM_RADIOS * MAX_NUM_VAP_PER_RADIO)]; /**< Interface map. */
+    radio_interface_mapping_t radio_interface_map[MAX_NUM_RADIOS]; /**< Radio interface map. */
+    BOOL radio_presence[MAX_NUM_RADIOS]; /**< Indicates if the interface is present (not in deep sleep). */
+    wifi_multi_link_info_t mu_info; /**< Multi-Link information. */
+    UINT BssMaxStaAllow; /**< Maximum number of stations supported for the given platform. Gets populated during bring-up. */
+} __attribute__((packed)) wifi_platform_property_t;
```

Comment:
These elements are missing in the wifi_platform_property_t structure 

   //Device Information related fields
   CHAR manufacturer[DEFAULT_DEVICE_FIELD_LEN];
   CHAR serialNo[DEFAULT_DEVICE_FIELD_LEN];
   CHAR manufacturerModel[DEFAULT_DEVICE_FIELD_LEN];
   CHAR software_version[DEFAULT_DEVICE_FIELD_LEN];
   mac_address_t cm_mac;
   mac_address_t al_1905_mac;

Reply:
ah you updated your branch after I pulled it, let me add these

578e6b13520f3b6274181f2859d93e8b5781d89c